### PR TITLE
TypeScript typings

### PR DESCRIPTION
**Pull request by niilo**
TypeScript typings for libvoikko v4.2

### CODE REVIEW

File: libvoikko/typescript/libvoikko.d.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
+export = Libvoikko;
+
+declare class Libvoikko {
+  constructor();
+  init(language: string): Voikko;
```
Comment (hatapitk) [inline libvoikko/typescript/libvoikko.d.ts]: Missing optional parameter here
```suggestion
  init(language: string, path?: string): Voikko;
```

File: libvoikko/typescript/libvoikko.d.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+}
+
+declare interface Analysis {
+  BASEFORM: string;
+  CLASS: string;
+  FSTOUTPUT: string;
+  NUMBER: string;
+  SIJAMUOTO: string;
+  STRUCTURE: string;
+  WORDBASES: string;
```
Comment (hatapitk) [inline libvoikko/typescript/libvoikko.d.ts]: These attributes are returned for nouns. There are far more attributes that can be returned for example for verbs. On the other hand it is not guaranteed that any attributes are present (especially if analyzing languages other than Finnish). Perhaps you could take all attributes listed in https://github.com/voikko/corevoikko/blob/master/libvoikko/doc/morphological-analysis.txt but mark them optional?

File: libvoikko/typescript/libvoikko.d.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+}
+
+declare interface Token {
+  type: TokenType;
+  text: string;
+}
+
+type TokenType = 'WORD' | 'WHITESPACE' | 'PUNCTUATION';
```
Comment (hatapitk) [inline libvoikko/typescript/libvoikko.d.ts]: Missing possible types:
```suggestion
type TokenType = 'WORD' | 'WHITESPACE' | 'PUNCTUATION' | 'NONE' | 'UNKNOWN';
```

File: libvoikko/typescript/libvoikko.d.ts [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+declare interface Sentence {
+  text: string;
+  nextStartType: NextStartType;
+}
+
+type NextStartType = 'PROBABLE' | 'NONE';
```
Comment (hatapitk) [inline libvoikko/typescript/libvoikko.d.ts]: Missing possible types:
```suggestion
type NextStartType = 'PROBABLE' | 'NONE' | 'POSSIBLE' | 'NO_START';
```

File: libvoikko/typescript/libvoikko.d.ts [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+   * ' ' = no hyphenation at this character,
+   * '-' = hyphenation point (character at this position is preserved in the hyphenated form),
+   * '=' = hyphentation point (character at this position is replaced by the hyphen.)
+   */
+  getHyphenationPattern: (word: string) => string;
+
+  /**
+   * Returns the given word in fully hyphenated form.
+   */
+  hyphenate: (word: string) => string;
```
Comment (hatapitk) [inline libvoikko/typescript/libvoikko.d.ts]: Missing optional parameters:
```suggestion
  hyphenate: (word: string, separator?: string, allowContextChanges?: boolean) => string;
```

File: libvoikko/typescript/libvoikko.d.ts (hunk 2 above)
Comment (niilo) [reply libvoikko/typescript/libvoikko.d.ts]: Sounds like a good plan. I'll add some documentation for those attributes.

**Comment by hatapitk** [comment]
Looks good, will merge this. Thanks for your work!

**Pull request merged**